iceoryx2-cal/tests/event_tests.rs

```diff
@@ -84,14 +84,41 @@ mod event {
         assert_that!(sut.err().unwrap(), eq NotifierCreateError::DoesNotExist);
     }
 
+    #[test]
+    fn notify_with_same_id_does_not_lead_to_non_blocking_timed_wait<Sut: Event>() {
+        let _watchdog = Watchdog::new();
+        const REPETITIONS: u64 = 8;
+        let name = generate_name();
+
+        let sut_listener = Sut::ListenerBuilder::new(&name).create().unwrap();
+        let sut_notifier = Sut::NotifierBuilder::new(&name).open().unwrap();
+
+        let trigger_id = TriggerId::new(0);
+
+        for _ in 0..REPETITIONS {
+            sut_notifier.notify(trigger_id).unwrap();
+        }
+
+        assert_that!(sut_listener.try_wait().unwrap(), is_some);
+
+        let now = Instant::now();
+        let result = sut_listener.timed_wait(TIMEOUT).unwrap();
+
+        if result.is_some() {
+            assert_that!(result, eq Some(trigger_id));
+        } else {
+            assert_that!(now.elapsed(), time_at_least TIMEOUT );
+        }
```

Comment:
Shouldn't that be handled a layer further down  and the concept behave the same?

In case a queue is used, the `timed_wait` could drain the queue and store the id's locally. Nothing for this PR but maybe a follow up for v0.4